modules/setting/queue.go

```diff
@@ -120,18 +120,28 @@ func NewQueueService() {
 				Indexer.IssueQueueType)
 		}
 	}
-	if _, ok := issueIndexerSectionMap["LENGTH"]; !ok {
+	if _, ok := sectionMap["LENGTH"]; !ok {
 		section.Key("LENGTH").SetValue(fmt.Sprintf("%d", Indexer.UpdateQueueLength))
 	}
-	if _, ok := issueIndexerSectionMap["BATCH_LENGTH"]; !ok {
+	if _, ok := sectionMap["BATCH_LENGTH"]; !ok {
 		section.Key("BATCH_LENGTH").SetValue(fmt.Sprintf("%d", Indexer.IssueQueueBatchNumber))
 	}
-	if _, ok := issueIndexerSectionMap["DATADIR"]; !ok {
+	if _, ok := sectionMap["DATADIR"]; !ok {
 		section.Key("DATADIR").SetValue(Indexer.IssueQueueDir)
 	}
-	if _, ok := issueIndexerSectionMap["CONN_STR"]; !ok {
+	if _, ok := sectionMap["CONN_STR"]; !ok {
 		section.Key("CONN_STR").SetValue(Indexer.IssueQueueConnStr)
 	}
+
+	// Handle the old mailer configuration
+	section = Cfg.Section("queue.mailer")
+	sectionMap = map[string]string{}
```

Comment:
Where does this `sectionMap` go after this? Seems like it's only needed for holding the value of `SEND_BUFFER_LEN`.

Reply:
So sectionMap is literally thrown away. It's simply there to check if a section of the ini file has a key. The problem is you can't use section.Has because then go-ini will defer to the default value of the parent section if the current section does not have that value set.

I could be cheaper by making it a `map[string]bool` (and the one above it - this might be worth it) or in this case specifically checking just for this one string but, I felt that consistency and simpler is better.

Reply:
```suggestion
	sectionMap = map[string]bool{}
```